### DISCUSSION TITLE
How to use the pre-open chrome window for wappdriver ?

### DISCUSSION THREAD

**Discussion by NaveenKumar004** [aahnik/wappdriver#46] [Q&A] [answered]
Everytime I run this program it opens a new window instead of using the pre-existing window which already has Whatsapp logged in. Is there any way to make the program use the pre-existing window? Or is there any other way around?

I'm a fresher to intermediate level of programming but, I do have some idea of modules, imports, etc. I havn't worked with chromedriver before. So if there were any silly question by me plz do forgive, but clarify my doubt 😅.

**Comment by aahnik** [comment, marked as answer]
yes it is possible to use a pre-existing chrome window. But this feature is somewhat hidden in `wappdriver` and not recommended to use .

------

**TLDR**

# The best thing is to close any pre-existing window before running code of wappdriver. Now everything will work as planned.

------

## Leave WhatsApp. Use Telegram.

or 

## Recommended
Dont use your personal WhatsApp account for sending messages with WappDriver. Create a seperate account. 

And now use WappDriver. You will need to scan the QR code only once. It will stay logged in. ( Unless you log-out or use the same  WhatsApp account in WhatsApp Web or Desktop app ). 

That means you will stay logged into two different WhatsApp accounts. One in your normal Chrome ( default user directory ) and another in WappDriver's chrome window. ( this session is stored inside the `~/wappdriver.data` folder .)

### Deep Understanding:
- When you login into any website, the session files get stored in **some folder** of your computer. When you open that website again, the same session is used, so you dont need to login again.
- The location of the **some folder** for Chrome, is different in different OS.
- This some folder is also called the "Chrome Profile Folder"
- The location for Chrome’s default profile folder differs depending on your platform. The locations are:
   -  Windows : `C:\Users\<username>\AppData\Local\Google\Chrome\User Data\Default`
    - Mac : `Users/<username>/Library/Application Support/Google/Chrome/Default`
    - Linux : `/home/<username>/.config/google-chrome/default`
- **Wappdriver does not use the default session folder of Chrome. It creates a new session folder inside `~/wappdriver.data/sessions`**

When you use `wappdriver` for the first time, it will require you to scan the QR code to login into WhatsApp. When you use it for the second time, it will automatically login. 

Suppose you have a WhatsApp in chrome window already open, then please close it first. WhatsApp Web cant run in two places at the same time.

Using the Chrome's default folder for wappdriver gives some unpredictable problems. So not recommended.

or

## Answer to your question

Open the file `~/wappdriver.data/selectors.yml` 

Now, add this line

```yml
user: "path/to/default/chrome/profile"
```
-------
Thank you for reading

-----
<details>
<summary> 👉 Click here to watch an ad </summary>

[![promoted](https://raw.githubusercontent.com/aahnik/aahnik/master/promoted/general.gif)](https://aahnik.github.io/aahnik/promoted/general.html)

</details>

**Reply by NaveenKumar004** [reply]
Thanks a lot. Your reply was very helpfull. I'm glad you spent time for me in your busy schedule for preparing your 12th boards.
How to send msgs in telegram with this same process, plz explain.
